services/QuillLMS/engines/evidence/lib/evidence/open_ai/embedding_fetcher.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Evidence
+  module OpenAI
+    class EmbeddingFetcher < Evidence::ApplicationService
+      include Evidence::OpenAI::Concerns::Api
+
+      ENDPOINT = '/embeddings'
+
+      attr_reader :dimension, :input, :model
+
+      def initialize(dimension:, input:, model:)
+        @dimension = dimension
+        @input = input
+        @model = model
+      end
+
+      def cleaned_results
+        response
+          .parsed_response['data']
+          &.first
```

Comment:
Do we really want safe nav operators here? This could result in some records having `nil` or `[]` embeddings without us easily knowing. 

If you prefer not throwing an exception on unparsed input, we could add a column to the new table in order to earmark records that need to be run through OpenAI again.